src/components/App/index.test.tsx

```diff
@@ -1,16 +1,56 @@
 import React from 'react';
 import { shallow } from 'enzyme';
+import { Store } from 'redux';
 
 import App from '.';
 import styles from './styles.module.scss';
 import configureStore from '../../configureStore';
+import { actions as apiActions } from '../../reducers/api';
 
-it('renders without crashing', () => {
-  const store = configureStore();
-  // TODO: Use shallowUntilTarget()
-  // https://github.com/mozilla/addons-code-manager/issues/15
-  const root = shallow(<App />, { context: { store } }).shallow();
+describe(__filename, () => {
+  type RenderParams = {
+    store?: Store;
+    authToken?: string | null;
+  };
 
-  expect(root).toHaveClassName(styles.container);
-  expect(root.find(styles.header)).toHaveLength(1);
+  const render = ({
+    store = configureStore(),
+    authToken = 'some-token',
+  }: RenderParams = {}) => {
+    // TODO: Use shallowUntilTarget()
+    // https://github.com/mozilla/addons-code-manager/issues/15
+    const root = shallow(<App authToken={authToken} />, {
+      context: { store },
+    }).shallow();
+
+    return root;
+  };
+
+  it('renders without crashing', () => {
+    const root = render();
+
+    expect(root).toHaveClassName(styles.container);
+    expect(root.find(styles.header)).toHaveLength(1);
+  });
+
+  it('dispatches setAuthToken on mount when authToken is valid', () => {
+    const authToken = 'my-token';
+    const store = configureStore();
+    const dispatch = jest.spyOn(store, 'dispatch');
+
+    render({ authToken, store });
+
+    expect(dispatch).toHaveBeenCalledWith(
+      apiActions.setAuthToken({ authToken }),
+    );
+  });
+
+  it('does not dispatch setAuthToken on mount when authToken is null', () => {
```

Comment:
I see you fixed the infinite loop, but could we have a test that would have caught that?

Reply:
Personally I think regression tests like that cause more harm than good because they are often slow and flaky, depending on how they are implemented it. They are hard to get right.